plugins/optimization-detective/class-od-url-metric-group.php

```diff
@@ -233,7 +235,12 @@ public function is_complete(): bool {
 			}
 			$current_time = microtime( true );
 			foreach ( $this->url_metrics as $url_metric ) {
-				if ( $current_time > $url_metric->get_timestamp() + $this->freshness_ttl ) {
+				if (
+					$current_time > $url_metric->get_timestamp() + $this->freshness_ttl
+					||
+					// If the URL Metric's ETag doesn't match the current ETag, consider the URL metric as stale.
+					( null !== $this->collection && ! hash_equals( $url_metric->get_etag() ?? md5( '' ), $this->collection->get_current_etag() ) )
```

Comment:
Aside: I just realized that there's not any reason for `collection` to not be required. I think it was only optional because of tests. So we can make this required, although this doesn't need to be done in this PR.

Reply:
Addressed in https://github.com/WordPress/performance/pull/1705/commits/5e395810d0cd791937ba6731473db6fd22e72707